CONFIGURATION.md

```diff
@@ -0,0 +1,297 @@
+
+`hlx publish` should fail when no `default` strain exists.
+`hlx up` can continue without a `default` strain, but should issue a warning
+
+> Notes:
+> - Option (1) is favoured.
+> - selecting a predefined strain is simple with yaml references, eg: `default: *strains.master`
+
+## Proposal: There is no implicit inheritance from `default`
+
+All inheritance becomes explicit, using the mechanism laid out in 
+https://github.com/adobe/helix-shared/issues/11#issuecomment-443122026. This means the `default` strain is special only in regards to serving traffic, but does not alter the behavior of any other strain.
+
+## Proposal: Strains can have a `mount` configuration that maps URL paths to content paths
+
+**DECLINED**
+
+> Notes:
+> We decided during the 4/2018 hackathon not to introduce mount points, so that it will be X strains for X mount points.
+
+<del>
+Right now, strains can have only one `content` node. With support for `mount`, a strain can have either a `content` or a `mount` property. Each `mount` object is a map between the URL path (key) and a `content` property (either git URL or object).
+
+```yaml
+strains:
+  default:
+    mount:
+      /: https://github.com/adobe/project-helix.io.git
+      /pipeline: https://github.com/adobe/helix-pipeline.git
+      /cli: https://github.com/adobe/helix-cli.git
+```
+</del>
+
+## Proposal: `mount` points can take Git URLs, `content` objects/references, Backend URLs, `origin` objects/references
+
+> Notes:
+> We decided during the 4/2018 hackathon not to introduce mount points, so that it will be X strains for X mount points.
+
+<del>
+That way you can have a consistent traffic mapping without having to switch strains.
+</del>
+
+## Discussion: the `code` property
+
+- the `code` property of a strain is either a giturl string or giturl object (but not an action name).
+
+assume a non-dirty checkout, `git@github.com/adobe/helix-io.git#master (923c1e6)`.
+1. the default strain contains a code property: `code: "git@github.com/adobe/helix-io.git#master"`
+2. `hlx deploy` would generate an action `/helix/923c1e6/html`
+3. and `effective-helix-config.yaml` would then have a property: `code: "git@github.com/adobe/helix-io.git#923c1e6"`
+
+assume a dirty checkout, `git@github.com/adobe/helix-io.git#master (a0c96a5)`
+1. the default strain contains a code property: `code: "git@github.com/adobe/helix-io.git#master"`
+2. `hlx deploy --dirty` would generate an action `/helix/a0c96a5-dirty/html`
+3. and `effective-helix-config.yaml` would then have a property: `code: "git@github.com/adobe/helix-io.git#a0c96a50-dirty"`<sup>1</sup>
+
+<sup>1</sup> not sure about the `-dirty` suffix in the ref here. but fastly wouldn't have a way to figure it our
+
+
+## Proposal: `hlx deploy` lists all strains that are affected by the deployment and suggests the creation of a new strain if none are affected
```

Comment:
I agree with Alex. `deploy` should probably still be a separate command for users who want to verify stuff on I/O Runtime before publishing, but it could be run implicitly by `publish` (if not run before).

Reply:
I think the separation of commands (see `hlx package`) is a good idea, but we should also run stuff for the user, so that a `hlx publish` in a fresh project starts a chain of `build`, `package`, `deploy`.

Reply:
I'd rather see that as "simple" vs "super advanced / specific usage": 90% of the users might not need / want to know about `build` `package` `deploy`...